File: app/src/am_hieunt/com/example/asian/issue11/api/ApiService.java
package com.example.asian.issue11.api;

import com.example.asian.issue11.model.Image;

import java.util.List;

import okhttp3.MultipartBody;
import okhttp3.Response;
import okhttp3.ResponseBody;
import retrofit2.Call;
import retrofit2.Retrofit;
import retrofit2.converter.gson.GsonConverterFactory;
import retrofit2.http.DELETE;
import retrofit2.http.GET;
import retrofit2.http.Header;
import retrofit2.http.Multipart;
import retrofit2.http.POST;
import retrofit2.http.Part;
import retrofit2.http.Path;

public interface ApiService {
    ApiService apiService = new Retrofit.Builder()
            .baseUrl("https://api.gyazo.com/api/")
            .addConverterFactory(GsonConverterFactory.create())
            .build()
            .create(ApiService.class);

    @GET("images")
    Call<List<Image>> getImages(@Header("Authorization") String token);

    @Multipart
    @POST("https://upload.gyazo.com/api/upload")
    Call<Image> uploadImage(@Header("Authorization") String token, @Part MultipartBody.Part imageData);

    @DELETE("images/{id}")
    Call<ResponseBody> deleteImage(@Header("Authorization") String token, @Path("id") String id);
}

File: app/src/am_hieunt/com/example/asian/issue13/ex2/api/ApiService.kt
package com.example.asian.issue13.ex2.api

import com.example.asian.issue13.ex2.model.Image
import retrofit2.Response
import retrofit2.Retrofit
import retrofit2.converter.gson.GsonConverterFactory
import retrofit2.http.GET
import retrofit2.http.Header

interface ApiService {
    companion object {
        var apiService: ApiService = Retrofit.Builder()
            .baseUrl("https://api.gyazo.com/api/")
            .addConverterFactory(GsonConverterFactory.create())
            .build()
            .create(ApiService::class.java)
    }

    @GET("images")
    suspend fun getImages(@Header("Authorization") token: String) : Response<List<Image>>
}
